lib/inc/drogon/HttpFilter.h

```diff
@@ -60,8 +62,38 @@ class HttpFilter : public DrObject<T>, public HttpFilterBase
 {
   public:
     static constexpr bool isAutoCreation{AutoCreation};
-    virtual ~HttpFilter()
+    ~HttpFilter() override = default;
+};
+
+#ifdef __cpp_impl_coroutine
+template <typename T, bool AutoCreation = true>
+class HttpCoroFilter : public DrObject<T>, public HttpFilterBase
+{
+  public:
+    static constexpr bool isAutoCreation{AutoCreation};
+    ~HttpCoroFilter() override = default;
+    void doFilter(const HttpRequestPtr &req,
+                  FilterCallback &&fcb,
+                  FilterChainCallback &&fccb) final
     {
+        drogon::async_run([this,
+                           req,
+                           fcb = std::move(fcb),
+                           fccb = std::move(fccb)]() -> drogon::Task<> {
+            HttpResponsePtr resp = co_await doFilter(req);
```

Comment:
What's the proper way to handle exceptions escaping from the doFilter coroutine. @marty1885